### PR TITLE
155 console errors

### PR DESCRIPTION
**Pull request by summer-cook**
co-author: @alishaevn 
co-author: @summer-cook 

# story
- fixes all the remaining errors in [#155](https://github.com/scientist-softserv/webstore/issues/155)

### CODE REVIEW

File: src/compounds/LinkedButton/LinkedButton.jsx [hunk 1]
```diff
@@ -31,10 +31,10 @@ const { onClick, ...remainingPropTypes } = Button.propTypes
 LinkedButton.propTypes = {
   buttonProps: PropTypes.shape(remainingPropTypes).isRequired,
   addClass: PropTypes.string,
-  path: PropTypes.exact({
+  path: PropTypes.oneOfType([PropTypes.shape({
     pathname: PropTypes.string.isRequired,
     query: PropTypes.shape({ id: PropTypes.number }),
-  }).isRequired,
+  }), PropTypes.string]).isRequired,
```
Comment (alishaevn) [inline src/compounds/LinkedButton/LinkedButton.jsx:37]: instead of allowing options here, can we update all instances of path to use the `{ pathname, query  }` format? this this code can remain as is and the rest of our code will be more uniform.

that can be pulled into a different pr though.

File: src/compounds/Messages/MessageCardBody.jsx [hunk 2]
```diff
@@ -52,7 +52,11 @@ MessageCardBody.propTypes = {
       }),
     ),
   }).isRequired,
-  isLatestMessage: PropTypes.bool.isRequired,
+  isLatestMessage: PropTypes.bool,
```
Comment (alishaevn) [inline src/compounds/Messages/MessageCardBody.jsx:55]: is this another change because of something loading?

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: src/compounds/Item/Item.jsx [hunk 3]
```diff
@@ -90,12 +90,12 @@ Item.propTypes = {
   isLoading: PropTypes.bool,
   item: PropTypes.shape({
     description: PropTypes.string,
-    id: PropTypes.number.isRequired,
+    id: PropTypes.number,
```
[the remark this answers is not in the document]
Comment (summer-cook) [reply src/compounds/Item/Item.jsx:93]: yes- the issue is that when the notice is rendered, it by default doesn't use the `item`.  So, we get a console error that the id is required, and there is no id.

File: src/compounds/Messages/MessageCardBody.jsx (hunk 2 above)
Comment (summer-cook) [reply src/compounds/Messages/MessageCardBody.jsx]: it was actually because this prop was just being used for a style and it wasn't really necessary, i just refactored and removed it altogether though because it was just removing padding for when it was the latest message. i checked it and it looks better with the padding anyways though

File: src/compounds/LinkedButton/LinkedButton.jsx (hunk 1 above)
Comment (summer-cook) [reply src/compounds/LinkedButton/LinkedButton.jsx]: i tried to do this and there is just one place where it is an issue. In the webstore, on the initiate request button for the make a request ware: `pages/requests/index.js`

this is how it started
image

i tried this and several other things. not sure if it will work since this is really a hardcoded link vs a dynamic query 
image

File: src/compounds/ItemGroup/ItemGroup.jsx [hunk 4]
```diff
@@ -56,7 +56,7 @@ ItemGroup.propTypes = {
     ...Item.propTypes,
     imgProps: PropTypes.shape({}),
     style: PropTypes.shape({}),
-  })).isRequired,
+  })),
```
[the remark this answers is not in the document]
Comment (summer-cook) [reply src/compounds/ItemGroup/ItemGroup.jsx]: this is a change in the webstore so it would be in a separate pr.

File: src/compounds/Item/Item.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (alishaevn) [reply src/compounds/Item/Item.jsx:93]: cool. let's write up a new ticket then to do my suggestion above instead por favor. 🙃

File: src/compounds/LinkedButton/LinkedButton.jsx (hunk 1 above)
Comment (alishaevn) [reply src/compounds/LinkedButton/LinkedButton.jsx]: try this:
``` js
path={{ pathname: `/requests/new/make-a-request?id=${defaultWareID}`, query: {} }}
```

File: src/compounds/ItemGroup/ItemGroup.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (alishaevn) [reply src/compounds/ItemGroup/ItemGroup.jsx]: ok. let's do that.

[the remark this answers is not in the document]
Comment (summer-cook) [reply src/compounds/ItemGroup/ItemGroup.jsx]: https://github.com/scientist-softserv/webstore/pull/165 heres a draft PR for it, but making this change brings back the same console error.

[the remark this answers is not in the document]
Comment (summer-cook) [reply src/compounds/ItemGroup/ItemGroup.jsx]: image

File: src/compounds/Item/Item.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (summer-cook) [reply src/compounds/Item/Item.jsx:93]: i tried to do something similar in the webstore to what you suggested below:
`item={service || {}}`
and the console error comes back:
image

File: src/compounds/LinkedButton/LinkedButton.jsx (hunk 1 above)
Comment (summer-cook) [reply src/compounds/LinkedButton/LinkedButton.jsx]: its so close... it won't load the page like this for some reason though. it is encoding the question mark to %3F, will look at if its possible to stop that from happening unless you have another suggestion 😕

image

Comment (summer-cook) [reply src/compounds/LinkedButton/LinkedButton.jsx]: got it- it was 

`path={{ pathname: `/requests/new/make-a-request`, query: { id: defaultWareID } }}`

pushed it up to https://github.com/scientist-softserv/webstore/pull/165

File: src/compounds/ItemGroup/ItemGroup.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (alishaevn) [reply src/compounds/ItemGroup/ItemGroup.jsx]: ref: https://github.com/scientist-softserv/webstore/pull/167

let me know if you wanna talk about it. 🙂

File: src/compounds/Item/Item.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (alishaevn) [reply src/compounds/Item/Item.jsx:93]: ref: https://github.com/scientist-softserv/webstore/pull/167 and #141.

let me know if you wanna talk about it. 🙂

File: src/compounds/Item/Item.jsx [hunk 5]
```diff
@@ -92,10 +92,10 @@ Item.propTypes = {
     description: PropTypes.string,
     id: PropTypes.number.isRequired,
     img: PropTypes.shape({
-      src: PropTypes.string.isRequired,
+      src: PropTypes.string,
       alt: PropTypes.string,
-    }).isRequired,
-    name: PropTypes.string.isRequired,
+    }),
```
Comment (alishaevn) [inline src/compounds/Item/Item.jsx:97]: why are we no longer requiring the img or the img src?

File: src/components/NextLink/NextLink.jsx [hunk 6]
```diff
@@ -23,7 +23,7 @@ NextLink.propTypes = {
   addClass: PropTypes.string,
   path: PropTypes.exact({
     pathname: PropTypes.string.isRequired,
-    query: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    query: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.shape({})]),
```
Comment (alishaevn) [inline src/components/NextLink/NextLink.jsx:26]:
```suggestion
    query: PropTypes.shape({}),
```

File: src/compounds/LinkedButton/LinkedButton.jsx [hunk 7]
```diff
@@ -31,7 +31,7 @@ const { onClick, ...remainingPropTypes } = Button.propTypes
 LinkedButton.propTypes = {
   buttonProps: PropTypes.shape(remainingPropTypes).isRequired,
   addClass: PropTypes.string,
-  path: PropTypes.exact({
+  path: PropTypes.shape({
```
Comment (alishaevn) [inline src/compounds/LinkedButton/LinkedButton.jsx:34]: why change this?

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: src/compounds/Item/Item.jsx (hunk 5 above)
Comment (summer-cook) [reply src/compounds/Item/Item.jsx:97]: was using the same logic from when we paired about this, they were throwing an error when the item was loading. just added them back though

File: src/compounds/LinkedButton/LinkedButton.jsx (hunk 7 above)
Comment (summer-cook) [reply src/compounds/LinkedButton/LinkedButton.jsx:34]: previously it threw an error with exact when we passed it a path with no query
image

but but since now we are always passing the pathname and the query it is fine to bring it back

Comment (summer-cook) [reply src/compounds/LinkedButton/LinkedButton.jsx]: i see there is still a linting issue with the LinkedButton, but i maded a separate ticket for it here
https://github.com/scientist-softserv/webstore-component-library/issues/143

File: src/compounds/Item/Item.jsx [hunk 8]
```diff
@@ -95,7 +95,7 @@ Item.propTypes = {
       src: PropTypes.string.isRequired,
       alt: PropTypes.string,
     }).isRequired,
-    name: PropTypes.string.isRequired,
+    name: PropTypes.string,
```
Comment (alishaevn) [inline src/compounds/Item/Item.jsx:98]: why would the name not be required?

File: src/compounds/Item/Item.jsx [hunk 9]
```diff
@@ -115,6 +115,13 @@ Item.defaultProps = {
   isLoading: false,
   item: {
     description: '',
+    id: null,
+    img: {
+      src: '',
+      alt: '',
+    },
```
Comment (alishaevn) [inline src/compounds/Item/Item.jsx:122]:
```suggestion
```

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: src/compounds/LinkedButton/LinkedButton.jsx [hunk 10]
```diff
@@ -1,7 +1,6 @@
 import React from 'react'
 import Link from 'next/link'
 import PropTypes from 'prop-types'
-import Button from '../../components/Button/Button'
```
Comment (alishaevn) [inline src/compounds/LinkedButton/LinkedButton.jsx:4]: we need to keep this.

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: src/compounds/Item/Item.jsx [hunk 11]
```diff
@@ -115,6 +115,11 @@ Item.defaultProps = {
   isLoading: false,
   item: {
     description: '',
+    id: null,
+    img: {
+      alt: '',
+    },
```
Comment (alishaevn) [inline src/compounds/Item/Item.jsx:121]: I see that it isn't, but the entire `img` prop should be required for this component. and since `img` is also required, we don't want to define any of it.

Comment (alishaevn) [reply src/compounds/Item/Item.jsx:121]:
```suggestion
```

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: src/compounds/ItemGroup/ItemGroup.jsx [hunk 12]
```diff
@@ -2,7 +2,7 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { Col, Row } from 'react-bootstrap'
 import Title from '../../components/Title/Title'
-import Item from '../Item/Item'
+import Item, { itemPropTypes} from '../Item/Item'
```
Comment (alishaevn) [inline src/compounds/ItemGroup/ItemGroup.jsx:5]:
```suggestion
import Item, { itemPropTypes } from '../Item/Item'
```

File: README.md [hunk 13]
```diff
@@ -25,6 +25,11 @@ Use NPM version `v8.19.2` and please do not mix it with Yarn.
   npm run storybook # starts the storybook app so that we can view/test the components in a UI
   ```
 - A new browser tab will automatically open to "http://localhost:6006/"
+
+#### Troubleshooting
+- If you see the following when starting the app: 
+  - `Error: error:0308010C:digital envelope routines::unsupported`
+  - set the following env variable in your local shell: `export NODE_OPTIONS=--openssl-legacy-provider`
```
Comment (alishaevn) [inline README.md:32]:
```suggestion
  - set the following env variable in your local shell: `export NODE_OPTIONS=--openssl-legacy-provider`

```
